src/Foundation/NSUrl.cs

```diff
@@ -55,7 +55,7 @@ public bool Equals (NSUrl? url)
 			if (Uri.TryCreate (url.AbsoluteString, UriKind.Absolute, out var uri))
 				return uri;
 			else
-				return new Uri (url.AbsoluteString, UriKind.Relative);
+				return url.AbsoluteString == null ? null : new Uri (url.AbsoluteString, UriKind.Relative);
```

Comment:
We can always use a ! (this is the implementation I did). There is no issue in this approach, since if you get null, you return null and the operator is stating that is nullable.